doc/Makefile

```diff
@@ -7,6 +7,32 @@ SPHINXOPTS    ?=
 SPHINXBUILD   ?= sphinx-build
 SOURCEDIR     = .
 BUILDDIR      = build
+GITDIR    = ../docs
+EXAMPLESDIR  = ../examples
+
+togit:
+	@cp -a build/html/ "$(GITDIR)"
+
+github:
+	@make html
+	@make togit
+
+theory:
+	@pandoc theory/glossary.tex -f latex -t rst -s --bibliography=theory/bibliography.bib -o theory/glossary.rst
+
+all-gh:
+	@make clean
+	@make theory
+	@make sphinx-gal
+	@make github
+
+gh-commit:
+	@cd "$(GITDIR)"; git add *.html generated/*.html;\
+	git add objects.inv; git add searchindex.js; \ git add _sources/*.txt; git add _sources/theory/*.txt; git add theory/*.html; \
+	git add gallery/*.html; git add _images/*.png; git add _sources/gallery/*.txt; git commit -m "Update docs"; git push origin
```

Comment:
Once we are in the repository corresponding to GH-pages, we can just commit with,
```
git commit -a -m "Update docs"
```
to include everything without `git add`, as whitelisting manually poses the risk of forgetting something (particularly for new files).